File: single-language-analysis/dataset-files/python/test_smell_022.py
import pytest

class TestCuckooHashTest:
    def test_smell_022(self, map, load, expected_capacity):
        print("Capacity: ", map.capacity(), " HashFuncs: ", map.get_num_hash_funcs(), " Buckets: ", map.get_num_buckets(), " Load: ", load)
        time = time.time()
        keys = []
        for i in range(int(map.capacity() * load)):
            keys.append(LongKey(random.randint(0, 2**63 - 1)))

        for i in range(len(keys)):
            key = keys[i]
            v = str(key.x)
            map.put(key, v)
            assert map.contains_key(key)
            assert v == map.get(key)

        for i in range(len(keys)):
            key = keys[i]
            v = str(key.x)
            assert v == map.get(key)

        assert expected_capacity == map.capacity()
        duration = time.time() - time
        print(duration, " ms")

File: single-language-analysis/dataset-files/java/TestSmell022.java
// Original URL: https://github.com/nguyenmv2/ChessAI/blob/61c7d27b9697f9e84af1cde6aa063b73a7c154d1/src/chess/hash/CuckooHashTest.java#L125-L150

public class TestSmell022 {
	@Test
	public void basicTest(CuckooHashLong<LongKey,String> map, double load, int expectedCapacity) {
		System.out.println("Capacity: " + map.capacity() + " HashFuncs: " + map.getNumHashFuncs() + " Buckets: " + map.getNumBuckets() + " Load: " + load);
		long time = System.currentTimeMillis();
		ArrayList<LongKey> keys = new ArrayList<LongKey>();
		for (int i = 0; i < map.capacity() * load; ++i) {
			keys.add(new LongKey(random.nextLong()));
		}
		
		for (int i = 0; i < keys.size(); ++i) {
			LongKey key = keys.get(i);
			String v = Long.toString(key.x);
			map.put(key, v);
			assertTrue(map.containsKey(key));
			assertEquals(v, map.get(key));
		}
		
		for (int i = 0; i < keys.size(); ++i) {
			LongKey key = keys.get(i);
			String v = Long.toString(key.x);
			assertEquals(v, map.get(key));
		}
		
		assertEquals(expectedCapacity, map.capacity());
		long duration = System.currentTimeMillis() - time;
		System.out.println(duration + " ms");
	}
}
